pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -5360,6 +5392,17 @@ protected CompletableFuture<Void> internalSetShadowTopic(List<String> shadowTopi
             return FutureUtil.failedFuture(new RestException(Status.PRECONDITION_FAILED,
                     "Cannot specify empty shadow topics, please use remove command instead."));
         }
+        try {
+            shadowTopics.forEach(shadowTopic -> {
+                if (TopicName.get(shadowTopic).isPartitioned()) {
+                    throw new RestException(Status.PRECONDITION_FAILED,
+                            "Couldn't set a partition of a topic as the shadow topic.");
+                }
+            });
+        } catch (RestException e) {
+            return FutureUtil.failedFuture(e);
+        }
```

Comment:
A trivial range for loop will be more clear that it won't need to break the `forEach` by throwing an exception first.

```java
        for (var shadowTopic : shadowTopics) {
            if (TopicName.get(shadowTopic).isPartitioned()) {
                return CompletableFuture.failedFuture(new RestException(Status.PRECONDITION_FAILED,
                        "Couldn't set a partition of a topic as the shadow topic."));
            }
        }
```

Reply:
Or the following code for FP style:

```java
        if (shadowTopics.stream().map(TopicName::get).anyMatch(TopicName::isPartitioned)) {
            return CompletableFuture.failedFuture(new RestException(Status.PRECONDITION_FAILED,
                    "Couldn't set a partition of a topic as the shadow topic."));
        }
```